### ISSUE TITLE
Generate error, Argument list too long.

### ISSUE THREAD

**Issue by o2e** [google/protobuf-gradle-plugin#633]
I have a huge project that contains 14,982 proto files and I get an "Argument list too long" error when I compile it. Is there any way to deal with it. I can't split this project at the moment, please help me or give me a suggestion. Thank you!

```
Execution failed for task ':generateProto'.
> java.io.IOException: Cannot run program "/Users/Smile/.gradle/caches/modules-2/files-2.1/com.google.protobuf/protoc/3.21.9/b5370f325a89e44a469213e65689c341f7d2a183/protoc-3.21.9-osx-x86_64.exe": error=7, Argument list too long
```

**Comment by rougsig** [comment]
Hello. Could you please provide a bit more details.

1 plugin version
2 used plugins
3 Windows version

**Comment by rougsig** [comment]
Also we have special logic for splitting long commands. Maybe sources can help to understand, what is going wrong. 

https://github.com/google/protobuf-gradle-plugin/blob/master/src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy#L178

It would be perfect if you can provide reproducible project.

**Comment by rougsig** [comment]
Could you please write absolute path size to your proto dir? Like `/Users/Smile/projects/myProject/app/src/main/proto` -> `50`

**Comment by rougsig** [comment]
Your proto files places in some directory. In argument in cli, plugin pass `<PATH_TO_PROTO_FILE>/myMessage.proto`. I want to known the length of `PATH_TO_PROTO_FILE`. Just to calculate how much approximately you have the length of the argument.

**Comment by karthikrg** [comment]
@rougsig My invocation is ~1.7MB long. It is a legacy project which I am converting from another IDL to proto with tight interdependencies between schemas, so it is kinda hard to split it up into multiple protoc invocations.

**Comment by karthikrg** [comment]
The cleaner solution actually seems to be to support the `@filename` option of protoc and append all args to the file. This would be similar to the approach adopted by the maven plugin in https://github.com/xolstice/protobuf-maven-plugin/pull/42/ . 

@rougsig would you be open to an intermediary hardcoded solution as proposed in #653 fast followed by a clean solution using `@filename`?

**Comment by rougsig** [comment]
@karthikrg Hardcoded solution is fine.

**Comment by rougsig** [comment]
@o2e yeah it will works too. Actually we already have splitting logic for Windows. For other OS it was disabled. Changes from @karthikrg enable that splitting.

**Comment by ejona86** [comment]
This should be fixed by #653
